test/test_treetensornetworks/test_solvers/test_dmrg.jl

```diff
@@ -108,35 +108,95 @@ end
-  s = siteinds("S=1/2", c)
 
-  os = ITensorNetworks.heisenberg(c)
+  @testset "Svd approach" for use_qns in [false, true]
+    s = siteinds("S=1/2", c; conserve_qns=use_qns)
 
-  H = TTN(os, s)
+    os = ITensorNetworks.heisenberg(c)
 
-  psi = random_ttn(s; link_space=20)
+    H = TTN(os, s)
 
-  nsweeps = 10
-  maxdim = [10, 20, 40, 100]
-  sweeps = Sweeps(nsweeps) # number of sweeps is 5
-  maxdim!(sweeps, 10, 20, 40, 100) # gradually increase states kept
-  cutoff!(sweeps, cutoff)
-  psi = dmrg(H, psi; nsweeps, maxdim, cutoff, nsite, solver_krylovdim=3, solver_maxiter=1)
+    # make init_state
+    d = Dict()
+    for (i, v) in enumerate(vertices(s))
+      d[v] = isodd(i) ? "Up" : "Dn"
+    end
+    states = v -> d[v]
+    psi = TTN(s, states)
 
-  # Compare to `ITensors.MPO` version of `dmrg`
-  linear_order = [4, 1, 2, 5, 3, 6]
-  vmap = Dictionary(vertices(s)[linear_order], 1:length(linear_order))
-  sline = only.(collect(vertex_data(s)))[linear_order]
-  Hline = MPO(relabel_sites(os, vmap), sline)
-  psiline = randomMPS(sline; linkdims=20)
-  e2, psi2 = dmrg(Hline, psiline, sweeps; outputlevel=0)
-
-  @test inner(psi', H, psi) ≈ inner(psi2', Hline, psi2) atol = 1e-5
+    #    psi = random_ttn(s; link_space=20) #FIXME: random_ttn broken for QN conserving case
+
+    nsweeps = 10
+    maxdim = [10, 20, 40, 100]
+    sweeps = Sweeps(nsweeps) # number of sweeps is 5
+    maxdim!(sweeps, 10, 20, 40, 100) # gradually increase states kept
+    cutoff!(sweeps, cutoff)
+    @show use_qns
+    psi = dmrg(H, psi; nsweeps, maxdim, cutoff, nsite, solver_krylovdim=3, solver_maxiter=1)
+
+    # Compare to `ITensors.MPO` version of `dmrg`
+    linear_order = [4, 1, 2, 5, 3, 6]
+    vmap = Dictionary(vertices(s)[linear_order], 1:length(linear_order))
+    sline = only.(collect(vertex_data(s)))[linear_order]
+    Hline = MPO(relabel_sites(os, vmap), sline)
+    psiline = randomMPS(sline, i -> isodd(i) ? "Up" : "Dn"; linkdims=20)
+    e2, psi2 = dmrg(Hline, psiline, sweeps; outputlevel=0)
+
+    @test inner(psi', H, psi) ≈ inner(psi2', Hline, psi2) atol = 1e-5
+  end
+end
+
+@testset "Tree DMRG for Fermions" for nsite in [2]
+  auto_fermion_enabled = ITensors.using_auto_fermion()
+  if !auto_fermion_enabled
+    ITensors.enable_auto_fermion()
+  end
+
+  cutoff = 1e-12
+  tooth_lengths = fill(2, 3)
+  c = named_comb_tree(tooth_lengths)
+
+  @testset "Svd approach" for use_qns in [true]
+    s = siteinds("Electron", c; conserve_qns=use_qns)
+    U = 2.0
+    t = 1.3
+    tp = 0.6
+    os = ITensorNetworks.hubbard(c; U=U, t=t, tp=tp)
+    H = TTN(os, s)
+
+    # make init_state
+    d = Dict()
+    for (i, v) in enumerate(vertices(s))
+      d[v] = isodd(i) ? "Up" : "Dn"
+    end
+    states = v -> d[v]
+    psi = TTN(s, states)
+
+    nsweeps = 10
+    maxdim = [10, 20, 40, 100]
+    sweeps = Sweeps(nsweeps) # number of sweeps is 5
+    maxdim!(sweeps, 10, 20, 40, 100) # gradually increase states kept
+    cutoff!(sweeps, cutoff)
```

Comment:
Let's avoid using the `Sweeps` object, we really should deprecate it at this point, the keyword argument syntax is preferred.